tensorflow/python/data/ops/readers.py

```diff
@@ -272,11 +275,16 @@ def __init__(self,
         "footer_bytes", footer_bytes)
     self._buffer_size = convert.optional_param_to_tensor(
         "buffer_size", buffer_size, _DEFAULT_READER_BUFFER_SIZE_BYTES)
+    self._compression_type = convert.optional_param_to_tensor(
+        "compression_type",
+        compression_type,
+        argument_default="",
+        argument_dtype=dtypes.string)
 
   def _as_variant_tensor(self):
-    return gen_dataset_ops.fixed_length_record_dataset(
+    return gen_dataset_ops.fixed_length_record_dataset_v2(
```

Comment:
This potentially breaks [forward compatibility](https://www.tensorflow.org/guide/version_compat#backward_and_partial_forward_compatibility) for existing users of `FixedLengthRecordDataset`. To fix, we need to do two things:

1. Detect when the `compression_type` is None (because all existing programs will have that property).
2. When `compression_type` is None, keep generating the v1 op. The typical way to do this is:

```python
if self._compression_type is not None or compat.forward_compatible(2018, 9, 30):
  # Use new code.
else: 
  # Use old code.
```